smartlearner/tasks/stopping_criteria.py

```diff
@@ -17,3 +19,31 @@ def __init__(self, nb_epochs_max):
     def post_epoch(self, status):
         if status.current_epoch >= self.nb_epochs_max:
             raise TrainingExit(status)
+
+
+class EarlyStopping(Task):
+    def __init__(self, cost, lookahead, callback=None, eps=0.):
+        super(EarlyStopping, self).__init__()
+
+        self.cost = cost
+        self.best_epoch = 0
+        self.best_cost = np.inf
+
+        self.lookahead = lookahead
+        self.callback = callback
+        self.eps = eps
+
+    def init(self, status):
+        self.best_cost = self.cost.view(status)
```

Comment:
If we don't do this, it means that at the end of epoch #1, we will necessarily find a better model (previous cost being infinite!) thus resetting the "epochs counter". You might think this is fine but let say you have a model that diverges from the start you expect early stopping to stop after `lookahead` epochs not `lookahead+1`.

Computing the cost at epoch #0 (which isn't a real epoch) allows us to effectively stop after `lookahead` epochs.